### PR TITLE
Drop-off fabric sensitive event when fabric is undefined

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
https://github.com/project-chip/connectedhomeip/issues/14366

#### Change overview
Drop-off fabric sensitive event when fabric is undefined

#### Testing
Add python testing for it

### CODE REVIEW

File: src/app/EventLogging.h [hunk 1]
```diff
@@ -73,6 +73,8 @@ CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventNumber & aE
     eventOptions.mPriority    = aEventData.GetPriorityLevel();
     eventOptions.mFabricIndex = aEventData.GetFabricIndex();
 
+    VerifyOrReturnError(eventOptions.mFabricIndex != kUndefinedFabricIndex, CHIP_NO_ERROR);
```
Comment (tcarmelveilleux) [inline src/app/EventLogging.h:76]: Suggest adding a comment that this skips logging the event if it's fabric-scoped but no fabric association exists yet.

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/EventLogging.h [hunk 2]
```diff
@@ -73,6 +73,8 @@ CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventNumber & aE
     eventOptions.mPriority    = aEventData.GetPriorityLevel();
     eventOptions.mFabricIndex = aEventData.GetFabricIndex();
```
Comment (tcarmelveilleux) [inline src/app/EventLogging.h:74]: Wonder if we should check that the event being reported is for a fabric that exists in fabric table

**Review by woody-apple — APPROVED** [review]

File: src/app/EventLogging.h (hunk 1 above)
Comment (yunhanw-google) [reply src/app/EventLogging.h:76]: done, thanks

Comment (yunhanw-google) [reply src/app/EventLogging.h:76]: would follow-up it, https://github.com/project-chip/connectedhomeip/issues/21982, thanks

File: src/app/EventLogging.h (hunk 2 above)
Comment (yunhanw-google) [reply src/app/EventLogging.h:74]: would follow-up it, https://github.com/project-chip/connectedhomeip/issues/21982, thanks

File: src/app/EventLogging.h [hunk 3]
```diff
@@ -72,6 +72,8 @@ CHIP_ERROR LogEvent(const T & aEventData, EndpointId aEndpoint, EventNumber & aE
     eventOptions.mPath        = path;
     eventOptions.mPriority    = aEventData.GetPriorityLevel();
     eventOptions.mFabricIndex = aEventData.GetFabricIndex();
+    // this skips logging the event if it's fabric-scoped but no fabric association exists yet.
+    VerifyOrReturnError(eventOptions.mFabricIndex != kUndefinedFabricIndex, CHIP_NO_ERROR);
```
Comment (mrjerryjohns) [inline src/app/EventLogging.h:76]: Why are we not returning an error here?

File: src/controller/python/test/test_scripts/cluster_objects.py [hunk 4]
```diff
@@ -331,6 +331,16 @@ async def TriggerAndWaitForEvents(cls, devCtrl, req):
...
         await cls._RetryForContent(request=lambda: devCtrl.ReadEvent(nodeid=NODE_ID, events=req), until=lambda res: res != 0)
 
+    @classmethod
+    @base.test_case
+    async def TestGenerateUndefinedFabricScopedEventRequests(cls, devCtrl):
+        logger.info("Running TestGenerateUndefinedFabricScopedEventRequests")
+        await devCtrl.SendCommand(nodeid=NODE_ID, endpoint=1, payload=Clusters.TestCluster.Commands.TestEmitTestFabricScopedEventRequest(arg1=0))
+        res = await devCtrl.ReadEvent(nodeid=NODE_ID, events=[
+            (1, Clusters.TestCluster.Events.TestEvent, 0),
+        ])
```
Comment (mrjerryjohns) [inline src/controller/python/test/test_scripts/cluster_objects.py:341]: What is this actually testing? If it isn't testing anything meaningful, we shouldn't have it here.

**Pull request merged**